Addrtide

webbjj opened the pull request:
Hi Jo

As promised, here is the pull request. Even though I had 4 commits, the first few were mostly me learning. The bulk of the changes were in the last commit and as you can see its not that extensive. Let me know if there are any issues.

Jeremy

jobovy commented on galpy/potential_src/Potential.py:
```diff
@@ -1510,6 +1510,166 @@ def nemo_accpars(self,vo,ro):
+        """
+        if self.isNonAxi:
+            #Make sure potential is axisymmetric
+            print("Error - cannot calculate rtide for non-axisymmetric potential")
```
Use a ``PotentialError`` here:
```
raise PotentialError("Cannot calculate rtide for non-axisymmetric potential")
```

jobovy commented on galpy/potential_src/Potential.py:
```diff
@@ -1510,6 +1510,166 @@ def nemo_accpars(self,vo,ro):
 
+    @potential_physical_input
+    @physical_conversion('position',pop=True)
+    def rtide(self,R,z,M=None,M_units=None):
```
Remove ``M_units``

jobovy commented on galpy/potential_src/Potential.py:
```diff
@@ -1510,6 +1510,166 @@ def nemo_accpars(self,vo,ro):
+            #If units not given, assume galpy! GM units?
+            print("Error - mass parameter has not been set")
+            return 0.0
+        else:
```
We should just assume the following two cases:

1. M is a scalar --> assumed to be in galpy units --> divide by bovy_conversion.mass_in_msol
2. M is a Quantity --> try units of mass or Gxmass

So I think this entire piece of code should be
```
### NO ELSE TO BEGIN
if _APY_LOADED and isinstance(M,units.Quantity):
    try:
        M= M.to(units.Msun).value/bovy_conversion.mass_in_msol(self._vo,self._ro)
    except units.UnitConversionError: pass
    else:
        try:
            M= M.to(units.pc*units.km**2/units.s**2).value/bovy_conversion.mass_in_msol(self._vo,self._ro)/bovy_conversion._G
        except units.UnitConversionError: pass
            raise units.UnitConversionError('M= parameter of rtide should have units of mass')
```

jobovy commented on galpy/potential_src/Potential.py:
```diff
@@ -1510,6 +1510,166 @@ def nemo_accpars(self,vo,ro):
+                        if not M_units == 'mass':
+                            raise units.UnitConversionError('M= parameter of %s should have units of %s, but has units of G x mass instead' % (type(self).__name__,M_units))
+                                    
+            #To calculate omegac in spherical coordinates we need the first derivative of the potential with respect to r
```
Switch back to the regular method indentation (you're doing this under the ``else`` clause here)

jobovy commented on galpy/potential_src/Potential.py:
```diff
@@ -1510,6 +1510,166 @@ def nemo_accpars(self,vo,ro):
+            Rderiv=-1.0*self.Rforce(R,z,use_physical=False)
+            zderiv=-1.0*self.zforce(R,z,use_physical=False)
+
+            omegac2=(Rderiv*(R/r)+zderiv*(z/r))/r
```
``Potential`` instances have a method ``rforce`` that gives the radial force (minus the derivative of the potential wrt spherical radius), isn't that just what you want to use here?

jobovy commented on galpy/potential_src/Potential.py:
```diff
@@ -1510,6 +1510,166 @@ def nemo_accpars(self,vo,ro):
+            z2deriv=self.z2deriv(R,z,use_physical=False)
+            Rzderiv=self.Rzderiv(R,z,use_physical=False)
+            
+            r2deriv=(R2deriv*nu.sin(theta)+Rzderiv*nu.cos(theta))*nu.sin(theta)+(Rzderiv*nu.sin(theta)+z2deriv*nu.cos(theta))*nu.cos(theta)
```
I think it would be good to introduce a new method ``r2deriv`` (like the way ``rforce`` is implemented) that gives the second derivative wrt the spherical radius and then use that here. This function might be useful more generally.

jobovy commented on galpy/potential_src/Potential.py:
```diff
@@ -1510,6 +1510,166 @@ def nemo_accpars(self,vo,ro):
+
+    @potential_physical_input
+    @physical_conversion('forcederivative',pop=True)
+    def ttensor(self,R,z,phi):
```
Make ``phi`` optional as ``phi=0.``  to allow this to be easily used with axisymmetric potentials

jobovy commented on galpy/potential_src/Potential.py:
```diff
@@ -1510,6 +1510,166 @@ def nemo_accpars(self,vo,ro):
+        """
+        if self.isNonAxi:
+            #Make sure potential is axisymmetric
+            print("Error - cannot calculate tidal tensor for non-axisymmetric potential")
```
Use ``PotentialError`` again

jobovy commented on galpy/potential_src/Potential.py:
```diff
@@ -1510,6 +1510,166 @@ def nemo_accpars(self,vo,ro):
+            #Make sure potential is axisymmetric
+            print("Error - cannot calculate tidal tensor for non-axisymmetric potential")
+            return 0.0
+        else:
```
No ``else`` here

jobovy commented on galpy/potential_src/Potential.py:
```diff
@@ -1510,6 +1510,166 @@ def nemo_accpars(self,vo,ro):
+    @potential_physical_input
+    @physical_conversion('forcederivative',pop=True)
+    def ttensor(self,R,z,phi):
+        """
```
Would it be useful to make it an option to return the eigenvalues instead? I.e., aren't we often interested in the eigenvalues?

jobovy commented on galpy/potential_src/Potential.py:
```diff
@@ -2755,3 +2915,164 @@ def kms_to_kpcGyrDecorator(func):
+
+@potential_physical_input
+@physical_conversion('position',pop=True)
+def rtide(Pot,R,z,M=None,M_units=None):
```
Basically all comments from ``rtide`` above apply to this function as well.

jobovy commented on galpy/potential_src/Potential.py:
```diff
@@ -2755,3 +2915,164 @@ def kms_to_kpcGyrDecorator(func):
+    """
+    if Pot.isNonAxi:
+        #Make sure potential is axisymmetric
+        print("Error - cannot calculate tidal tensor for non-axisymmetric potential")
```
Use ``PotentialError`` instead

webbjj commented:
Ah shoot, I actually committed too soon. Sorry. Left in a comment that I
wanted to ask you about. Pretty sure I do not need to do the axisymmetric
check for rtide since we just need rforce and r2deriv, which shouldn't
depend on whether or not the potential is axisymmetric. It just matters
what the values are at the cluster's current position because the
calculation itself assumes the cluster has a circular orbit its
instantaneous position. For ttensor though the check is still necessary. In
time / if necessary I could add the phi dependence to ttensor.

Jeremy

On Wed, Mar 21, 2018 at 1:35 PM, Jeremy Webb <webb.jjw@gmail.com> wrote:

> Hi Jo
>
> OK, think I addressed everything you asked and cleaned up a few more
> things. Let me know if there are any other issues.
>
> One quick note - I did add the option to return the eigenvalues of the
> tidal tensor. However, I think the key eigenvalues, that is the ones that
> allow for a calculation of rtide, are of the effective tidal tensor which
> is ttensor (which we calculate) - tensor of potential due to centrifugal
> acceleration. (See Equations 6-9 in http://adsabs.harvard.edu/
> abs/2011MNRAS.418..759R if you are interested). Ill have to think more
> about if we can calculate the effective tidal tensor for arbitrary cases.
> In the paper, Florent just does it for idealized cases. However this
> probably would be most useful. We could then just use ttensor to calculate
> rtide every time.
>
> Jeremy
>
> On Mon, Mar 19, 2018 at 5:42 PM, Jo Bovy <notifications@github.com> wrote:
>
>> *@jobovy* requested changes on this pull request.
>>
>> Looks like a great start! I've made a bunch of comments that I think will
>> improve the code and make it more consistent with the rest of galpy.
>>
>> If you keep pushing to your addrtide branch, commits will be
>> automatically added to this PR. Let me know when you're done with these,
>> and then I will merge in the latest master changes so the tests can run
>> (they can't run right now because the PR can't be automatically merged).
>> Make sure that you've ticked the box "Allow commits from maintainers",
>> otherwise I can't push to your branch.
>> ------------------------------
>>
>> In galpy/potential_src/Potential.py
>> <https://github.com/jobovy/galpy/pull/334#discussion_r175591561>:
>>
>> > +            z - height (can be Quantity)
>> +
>> +            M - (default = None) Mass of object (can be Quantity)
>> +
>> +        OUTPUT:
>> +
>> +            Tidal Radius
>> +
>> +        HISTORY:
>> +
>> +            2018-02-12 - Written - Webb (UofT)
>> +
>> +        """
>> +        if self.isNonAxi:
>> +            #Make sure potential is axisymmetric
>> +            print("Error - cannot calculate rtide for non-axisymmetric potential")
>>
>> Use a PotentialError here:
>>
>> raise PotentialError("Cannot calculate rtide for non-axisymmetric potential")
>>
>> ------------------------------
>>
>> In galpy/potential_src/Potential.py
>> <https://github.com/jobovy/galpy/pull/334#discussion_r175591752>:
>>
>> > @@ -1510,6 +1510,166 @@ def nemo_accpars(self,vo,ro):
>>          except AttributeError:
>>              raise AttributeError('NEMO acceleration parameters not supported for %s' % self.__class__.__name__)
>>
>> +    @potential_physical_input
>> +    @physical_conversion('position',pop=True)
>> +    def rtide(self,R,z,M=None,M_units=None):
>>
>> Remove M_units
>> ------------------------------
>>
>> In galpy/potential_src/Potential.py
>> <https://github.com/jobovy/galpy/pull/334#discussion_r175593208>:
>>
>> > +            Tidal Radius
>> +
>> +        HISTORY:
>> +
>> +            2018-02-12 - Written - Webb (UofT)
>> +
>> +        """
>> +        if self.isNonAxi:
>> +            #Make sure potential is axisymmetric
>> +            print("Error - cannot calculate rtide for non-axisymmetric potential")
>> +            return 0.0
>> +        elif M==None:
>> +            #If units not given, assume galpy! GM units?
>> +            print("Error - mass parameter has not been set")
>> +            return 0.0
>> +        else:
>>
>> We should just assume the following two cases:
>>
>>    1. M is a scalar --> assumed to be in galpy units --> divide by
>>    bovy_conversion.mass_in_msol
>>    2. M is a Quantity --> try units of mass or Gxmass
>>
>> So I think this entire piece of code should be
>>
>> ### NO ELSE TO BEGIN
>> if _APY_LOADED and isinstance(M,units.Quantity):
>>     try:
>>         M= M.to(units.Msun).value/bovy_conversion.mass_in_msol(self._vo,self._ro)
>>     except units.UnitConversionError: pass
>>     else:
>>         try:
>>             M= M.to(units.pc*units.km**2/units.s**2).value/bovy_conversion.mass_in_msol(self._vo,self._ro)/bovy_conversion._G
>>         except units.UnitConversionError: pass
>>             raise units.UnitConversionError('M= parameter of rtide should have units of mass')
>>
>> ------------------------------
>>
>> In galpy/potential_src/Potential.py
>> <https://github.com/jobovy/galpy/pull/334#discussion_r175593282>:
>>
>> > +                    if not M_units == 'mass':
>> +                        raise units.UnitConversionError('M= parameter of %s should have units of %s, but has units of mass instead' % (type(self).__name__,M_units))
>> +                if not unitFound:
>> +                    # G x mass
>> +                    try:
>> +                        M= M.to(units.pc*units.km**2/units.s**2)\
>> +                            .value\
>> +                            /bovy_conversion.mass_in_msol(self._vo,self._ro)\
>> +                            /bovy_conversion._G
>> +                    except units.UnitConversionError: pass
>> +                    else:
>> +                        unitFound= True
>> +                        if not M_units == 'mass':
>> +                            raise units.UnitConversionError('M= parameter of %s should have units of %s, but has units of G x mass instead' % (type(self).__name__,M_units))
>> +
>> +            #To calculate omegac in spherical coordinates we need the first derivative of the potential with respect to r
>>
>> Switch back to the regular method indentation (you're doing this under
>> the else clause here)
>> ------------------------------
>>
>> In galpy/potential_src/Potential.py
>> <https://github.com/jobovy/galpy/pull/334#discussion_r175593604>:
>>
>> > +                            /bovy_conversion.mass_in_msol(self._vo,self._ro)\
>> +                            /bovy_conversion._G
>> +                    except units.UnitConversionError: pass
>> +                    else:
>> +                        unitFound= True
>> +                        if not M_units == 'mass':
>> +                            raise units.UnitConversionError('M= parameter of %s should have units of %s, but has units of G x mass instead' % (type(self).__name__,M_units))
>> +
>> +            #To calculate omegac in spherical coordinates we need the first derivative of the potential with respect to r
>> +            r=math.sqrt(R**2.0+z**2.0)
>> +            theta=nu.arctan2(R, z)
>> +
>> +            Rderiv=-1.0*self.Rforce(R,z,use_physical=False)
>> +            zderiv=-1.0*self.zforce(R,z,use_physical=False)
>> +
>> +            omegac2=(Rderiv*(R/r)+zderiv*(z/r))/r
>>
>> Potential instances have a method rforce that gives the radial force
>> (minus the derivative of the potential wrt spherical radius), isn't that
>> just what you want to use here?
>> ------------------------------
>>
>> In galpy/potential_src/Potential.py
>> <https://github.com/jobovy/galpy/pull/334#discussion_r175593890>:
>>
>> > +
>> +            #To calculate omegac in spherical coordinates we need the first derivative of the potential with respect to r
>> +            r=math.sqrt(R**2.0+z**2.0)
>> +            theta=nu.arctan2(R, z)
>> +
>> +            Rderiv=-1.0*self.Rforce(R,z,use_physical=False)
>> +            zderiv=-1.0*self.zforce(R,z,use_physical=False)
>> +
>> +            omegac2=(Rderiv*(R/r)+zderiv*(z/r))/r
>> +
>> +            #To calculate the epicyclic frequency kappa we need the second derivative of the potential with respect to r
>> +            R2deriv= self.R2deriv(R,z,use_physical=False)
>> +            z2deriv=self.z2deriv(R,z,use_physical=False)
>> +            Rzderiv=self.Rzderiv(R,z,use_physical=False)
>> +
>> +            r2deriv=(R2deriv*nu.sin(theta)+Rzderiv*nu.cos(theta))*nu.sin(theta)+(Rzderiv*nu.sin(theta)+z2deriv*nu.cos(theta))*nu.cos(theta)
>>
>> I think it would be good to introduce a new method r2deriv (like the way
>> rforce is implemented) that gives the second derivative wrt the
>> spherical radius and then use that here. This function might be useful more
>> generally.
>> ------------------------------
>>
>> In galpy/potential_src/Potential.py
>> <https://github.com/jobovy/galpy/pull/334#discussion_r175594328>:
>>
>> > +
>> +            #To calculate the epicyclic frequency kappa we need the second derivative of the potential with respect to r
>> +            R2deriv= self.R2deriv(R,z,use_physical=False)
>> +            z2deriv=self.z2deriv(R,z,use_physical=False)
>> +            Rzderiv=self.Rzderiv(R,z,use_physical=False)
>> +
>> +            r2deriv=(R2deriv*nu.sin(theta)+Rzderiv*nu.cos(theta))*nu.sin(theta)+(Rzderiv*nu.sin(theta)+z2deriv*nu.cos(theta))*nu.cos(theta)
>> +
>> +            #rt=(GM/(omegac2*nu))**(1/3)
>> +            kappa2=3.0*omegac2+r2deriv
>> +            nuu=4.0-kappa2/omegac2
>> +            return (M/(omegac2*nuu))**(1.0/3.0)
>> +
>> +    @potential_physical_input
>> +    @physical_conversion('forcederivative',pop=True)
>> +    def ttensor(self,R,z,phi):
>>
>> Make phi optional as phi=0. to allow this to be easily used with
>> axisymmetric potentials
>> ------------------------------
>>
>> In galpy/potential_src/Potential.py
>> <https://github.com/jobovy/galpy/pull/334#discussion_r175594400>:
>>
>> > +
>> +            z - height (can be Quantity)
>> +
>> +
>> +        OUTPUT:
>> +
>> +            Tidal Tensor
>> +
>> +        HISTORY:
>> +
>> +            2018-03-07 - Written - Webb (UofT)
>> +
>> +        """
>> +        if self.isNonAxi:
>> +            #Make sure potential is axisymmetric
>> +            print("Error - cannot calculate tidal tensor for non-axisymmetric potential")
>>
>> Use PotentialError again
>> ------------------------------
>>
>> In galpy/potential_src/Potential.py
>> <https://github.com/jobovy/galpy/pull/334#discussion_r175594449>:
>>
>> > +
>> +
>> +        OUTPUT:
>> +
>> +            Tidal Tensor
>> +
>> +        HISTORY:
>> +
>> +            2018-03-07 - Written - Webb (UofT)
>> +
>> +        """
>> +        if self.isNonAxi:
>> +            #Make sure potential is axisymmetric
>> +            print("Error - cannot calculate tidal tensor for non-axisymmetric potential")
>> +            return 0.0
>> +        else:
>>
>> No else here
>> ------------------------------
>>
>> In galpy/potential_src/Potential.py
>> <https://github.com/jobovy/galpy/pull/334#discussion_r175594805>:
>>
>> > +            #To calculate the epicyclic frequency kappa we need the second derivative of the potential with respect to r
>> +            R2deriv= self.R2deriv(R,z,use_physical=False)
>> +            z2deriv=self.z2deriv(R,z,use_physical=False)
>> +            Rzderiv=self.Rzderiv(R,z,use_physical=False)
>> +
>> +            r2deriv=(R2deriv*nu.sin(theta)+Rzderiv*nu.cos(theta))*nu.sin(theta)+(Rzderiv*nu.sin(theta)+z2deriv*nu.cos(theta))*nu.cos(theta)
>> +
>> +            #rt=(GM/(omegac2*nu))**(1/3)
>> +            kappa2=3.0*omegac2+r2deriv
>> +            nuu=4.0-kappa2/omegac2
>> +            return (M/(omegac2*nuu))**(1.0/3.0)
>> +
>> +    @potential_physical_input
>> +    @physical_conversion('forcederivative',pop=True)
>> +    def ttensor(self,R,z,phi):
>> +        """
>>
>> Would it be useful to make it an option to return the eigenvalues
>> instead? I.e., aren't we often interested in the eigenvalues?
>> ------------------------------
>>
>> In galpy/potential_src/Potential.py
>> <https://github.com/jobovy/galpy/pull/334#discussion_r175594947>:
>>
>> > @@ -2755,3 +2915,164 @@ def kms_to_kpcGyrDecorator(func):
>>      def kms_to_kpcGyr_wrapper(*args,**kwargs):
>>          return func(args[0],velocity_in_kpcGyr(args[1],1.),args[2],**kwargs)
>>      return kms_to_kpcGyr_wrapper
>> +
>> +@potential_physical_input
>> +@physical_conversion('position',pop=True)
>> +def rtide(Pot,R,z,M=None,M_units=None):
>>
>> Basically all comments from rtide above apply to this function as well.
>> ------------------------------
>>
>> In galpy/potential_src/Potential.py
>> <https://github.com/jobovy/galpy/pull/334#discussion_r175595053>:
>>
>> > +        z - height (can be Quantity)
>> +
>> +        phi - azimuth (can be Quantity)
>> +
>> +    OUTPUT:
>> +
>> +        Tidal Tensor
>> +
>> +    HISTORY:
>> +
>> +        2018-03-19 - Written - Webb (UofT)
>> +
>> +    """
>> +    if Pot.isNonAxi:
>> +        #Make sure potential is axisymmetric
>> +        print("Error - cannot calculate tidal tensor for non-axisymmetric potential")
>>
>> Use PotentialError instead
>>
>> —
>> You are receiving this because you authored the thread.
>> Reply to this email directly, view it on GitHub
>> <https://github.com/jobovy/galpy/pull/334#pullrequestreview-105146455>,
>> or mute the thread
>> <https://github.com/notifications/unsubscribe-auth/AH4EYMGACw1Rw5kZ76KvN6qI5-FIv8Rgks5tgCYpgaJpZM4SwvE4>
>> .
>>
>
>

jobovy commented on galpy/potential_src/Potential.py:
```diff
@@ -389,7 +389,46 @@ def rforce(self,R,z,phi=0.,t=0.):
+        """
+        NAME:
+
+           rderiv
```
--> r2deriv

jobovy commented on galpy/potential_src/Potential.py:
```diff
@@ -1539,73 +1580,49 @@ def rtide(self,R,z,M=None,M_units=None):
+            2018-03-21 - Written - Webb (UofT)
             
         """
+        #DO WE NEED THIS? PHI IS WORKED INTO RFORCE AND R2DERIV?
```
If the result is sensible when just evaluating the forces and derivatives at R,z,phi,t then I agree that we don't need to do the check, so you can remove it.

jobovy commented on galpy/potential_src/Potential.py:
```diff
@@ -1539,73 +1580,49 @@ def rtide(self,R,z,M=None,M_units=None):
-            #If units not given, assume galpy! GM units?
-            print("Error - mass parameter has not been set")
+            #Make sure an object mass is given
+            print("RTIDE Error - mass parameter has not been set")
```
Also raise PotentialError here instead

jobovy commented on galpy/potential_src/Potential.py:
```diff
@@ -1619,55 +1636,59 @@ def ttensor(self,R,z,phi):
         if self.isNonAxi:
             #Make sure potential is axisymmetric
-            print("Error - cannot calculate tidal tensor for non-axisymmetric potential")
+            raise PotentialError("Cannot calculate ttensor for non-axisymmetric potential")
```
So again everything is just computed based on local forces and second derivatives, so why can this not be done for a non-axisymmetric potential?

jobovy commented on galpy/potential_src/Potential.py:
```diff
@@ -2918,100 +2939,79 @@ def kms_to_kpcGyr_wrapper(*args,**kwargs):
-        #If units not given, assume galpy! GM units?
-        print("Error - mass parameter has not been set")
+        #Make sure an object mass is given
+        print("RTIDE Error - mass parameter has not been set")
```
PotentialError here too.

webbjj commented:
Hi Jo

As you will likely see I made another commit. I believe this address all
your comments and I added a phi component to ttensor, but set zphideriv=0.0
since as we discussed this doesn't exist yet.

Ive attached my derivation of the tidal tensor components in case you want
to check it. The x and y derivatives I was able to cross check with your
planar_orbit.py file. The xz, yz, zx, and zy derivates I wasn't able to
cross check but they weren't terribly hard (I don't think!).

Let me know if you have more comments.

Jeremy

On Wed, Mar 21, 2018 at 2:03 PM, Jo Bovy <notifications@github.com> wrote:

> *@jobovy* requested changes on this pull request.
>
> Looks much better now, think I answered your questions (and asked some
> more!) in the review comments.
> ------------------------------
>
> In galpy/potential_src/Potential.py
> <https://github.com/jobovy/galpy/pull/334#discussion_r176181953>:
>
> > @@ -389,7 +389,46 @@ def rforce(self,R,z,phi=0.,t=0.):
>          r= nu.sqrt(R**2.+z**2.)
>          return self.Rforce(R,z,phi=phi,t=t,use_physical=False)*R/r\
>              +self.zforce(R,z,phi=phi,t=t,use_physical=False)*z/r
> +
> +    @potential_physical_input
> +    @physical_conversion('forcederivative',pop=True)
> +    def r2deriv(self,R,z,phi=0.,t=0.):
> +        """
> +        NAME:
> +
> +           rderiv
>
> --> r2deriv
> ------------------------------
>
> In galpy/potential_src/Potential.py
> <https://github.com/jobovy/galpy/pull/334#discussion_r176182444>:
>
> >
>          """
> +        #DO WE NEED THIS? PHI IS WORKED INTO RFORCE AND R2DERIV?
>
> If the result is sensible when just evaluating the forces and derivatives
> at R,z,phi,t then I agree that we don't need to do the check, so you can
> remove it.
> ------------------------------
>
> In galpy/potential_src/Potential.py
> <https://github.com/jobovy/galpy/pull/334#discussion_r176182712>:
>
> >              return 0.0
>          elif M==None:
> -            #If units not given, assume galpy! GM units?
> -            print("Error - mass parameter has not been set")
> +            #Make sure an object mass is given
> +            print("RTIDE Error - mass parameter has not been set")
>
> Also raise PotentialError here instead
> ------------------------------
>
> In galpy/potential_src/Potential.py
> <https://github.com/jobovy/galpy/pull/334#discussion_r176183295>:
>
> >          """
>          if self.isNonAxi:
>              #Make sure potential is axisymmetric
> -            print("Error - cannot calculate tidal tensor for non-axisymmetric potential")
> +            raise PotentialError("Cannot calculate ttensor for non-axisymmetric potential")
>
> So again everything is just computed based on local forces and second
> derivatives, so why can this not be done for a non-axisymmetric potential?
> ------------------------------
>
> In galpy/potential_src/Potential.py
> <https://github.com/jobovy/galpy/pull/334#discussion_r176183372>:
>
> >          return 0.0
>      elif M==None:
> -        #If units not given, assume galpy! GM units?
> -        print("Error - mass parameter has not been set")
> +        #Make sure an object mass is given
> +        print("RTIDE Error - mass parameter has not been set")
>
> PotentialError here too.
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/jobovy/galpy/pull/334#pullrequestreview-105840587>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AH4EYOoNkn0f18Nkt90hsm0LbXvhCcqXks5tgpXtgaJpZM4SwvE4>
> .
>
